sonar-core/src/main/resources/org/sonar/core/component/db/ComponentMapper.xml

```diff
@@ -227,5 +227,30 @@
     delete from projects where kee=#{key}
   </delete>
 
-</mapper>
+  <select id="selectProvisionedProjects" parameterType="map" resultType="Component">
+    select <include refid="componentColumns"/>
+    from projects p
+    <include refid="provisionClauses"/>
+  </select>
 
+  <select id="countProvisionedProjects" parameterType="map" resultType="int">
+    select count(p.id)
+    from projects p
+    <include refid="provisionClauses"/>
+  </select>
+
+  <sql id="provisionClauses">
+    left join snapshots s on s.project_id=p.id
+    where
+    s.id is null
```

Comment:
shouldn't "s.is is null" be moved to the join statement for clarity?
(not sure, I'm not very much used to the old/stantard SQL join syntax)